File: go/src/main/mrcoordinator.go
package main

import (
	"fmt"
	"io"
	"log"
	"mapreduce/mr"
	"os"
	"time"
)

func main() {
	log.SetOutput(io.Discard)
	if len(os.Args) < 2 {
		fmt.Fprintf(os.Stderr, "Usage: mrcoordinator inputfiles...\n")
		os.Exit(1)
	}
	m := mr.MakeCoordinator(os.Args[1:], 10)
	for m.Done() == false {
		time.Sleep(time.Second)
	}
	time.Sleep(time.Second)
}

File: c/src/main/mrcoordinator.c
#include <stdio.h>
#include <threads.h>
#include <time.h>

#include "../mr/coordinator.h"
#include "../util/err.h"

int main(int argc, char **argv) {
    if (argc < 2) err("Usage: mrcoordinator inputfiles...");
    Coord *c = makeCoord(&argv[1], argc - 1, 10);
    while (!done(c)) {
        if (thrd_sleep(&(struct timespec){.tv_sec = 1}, NULL))
            err("main (mrcoordinator): thrd_sleep failed");
    }
    if (thrd_sleep(&(struct timespec){.tv_sec = 1}, NULL))
        err("main (mrcoordinator): thrd_sleep failed");
}
